test/integration/smoke/test_vm_extra_config_data.py

```diff
@@ -0,0 +1,299 @@
+
+        host_id = start_resp.hostid
+        host = list_hosts(
+            self.apiclient,
+            id=host_id,
+            hypervisor='kvm')
+
+        instance_name = response.instancename
+        host_ipaddress = host[0].ipaddress
+
+        ssh_client = SshClient(host_ipaddress, port=22,
+                               user=self.hostConfig['username'],
+                               passwd=self.hostConfig['password'])
+        virsh_cmd = 'virsh dumpxml %s' % instance_name
+        xml_res = ssh_client.execute(virsh_cmd)
+        xml_as_str = ''.join(xml_res)
+
+        extraconfig_decoded_xml = '<config>' + urllib.unquote(extraconfig) + '</config>'
+
+        # Root XML Elements
+        parser = etree.XMLParser(remove_blank_text=True)
+        domain_xml_root = ET.fromstring(xml_as_str, parser=parser)
+        decoded_xml_root = ET.fromstring(extraconfig_decoded_xml, parser=parser)
+        for child in decoded_xml_root:
+            find_element_in_domain_xml = domain_xml_root.find(child.tag)
+
+            # Fail if extra config is not found in domain xml
+            self.assertNotEquals(
+                0,
+                len(find_element_in_domain_xml),
+                'Element tag from extra config not added to VM'
+            )
+
+            # Compare found XML node with extra config node
+            is_a_match = self.elements_equal(child, find_element_in_domain_xml)
+            self.assertEquals(
+                True,
+                is_a_match,
+                'The element from tags from extra config do not match with those found in domain xml'
+            )
+
+        self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_vmware(self):
+        if self.hypervisor.lower() != 'vmware':
+            raise self.skipTest("Skipping test case for non-vmware hypervisor")
+
+        extraconfig = None
+
+
+        response = self.deploy_vm(extraconfig)
+
+        host_id = response.hostid
+        host = list_hosts(
+            self.apiclient,
+            id=host_id)
+
+        instance_name = response.instancename
+        host_ipaddress = host[0].ipaddress
+
+        ssh_client = SshClient(host_ipaddress, port=22,
+                               user=self.hostConfig['username'],
+                               passwd=self.hostConfig['password'])
+
+        extraconfig_decoded = urllib.unquote(extraconfig)
+        config_arr = extraconfig_decoded.split("\\r?\\n")
+
+        for config in config_arr:
+            grep_config = "cat /vmfs/volumes/eaa90eaf-b789d201/%s/%s.vmx | grep -w '%s'" % (instance_name, instance_name, config)
+            result = ssh_client.execute(grep_config)
+            self.assertNotEquals(
+                0,
+                len(result),
+                'Extra  configuration not found in instance vmx file'
+            )
+
+        self.destroy_vm(response.id)
```

Comment:
I think we're missing a test for Xen Server @dhlaluku

Reply:
Let me add them quickly